sympy/matrices/matrices.py

```diff
@@ -3433,6 +3433,93 @@ def _eval_matrix_exp_jblock(self):
         from .sparsetools import banded
         return self.__class__(banded(size, bands))
 
+
+
+    def analytic_func(self, f, x, parameter = None):
+        """
+        Computes f(A) where A is a Square Matrix
+        and F is an univariate analytic function.
+
+        Examples
+        ========
+
+        >>> from sympy import Symbol, Matrix, exp, S, log
+
+        >>> x = Symbol('x')
+        >>> m = Matrix([[S(5)/4, S(3)/4], [S(3)/4, S(5)/4]])
+        >>> f = log(x)
+        >>> m.analytic_func(f, x)
+        Matrix([
+        [     0, log(2)],
+        [log(2),      0]])
+
+        Parameters
+        ==========
+
+        f : Univariate Analytic Function
+        x : parameter of f
+        parameter : If Symbol('t') then it returns f(A*t) where t is parameter
+
+        """
+        from sympy import eye, zeros, diff, solve
+
+        t = 1
+        if parameter:
+            t = sympify(parameter)
+            if t in self.free_symbols:
+                raise ValueError("The parameter should not be in A")
+        f_0 = f
+        f = f.subs(x, x*t)
+        eigen = self.eigenvals()
+
+        max_mul = max(eigen.values())
+        derivative = {}
+        dd = f
+        for i in range(max_mul - 1):
+            dd= diff(dd, x)
+            derivative[i + 1] = dd
+        n = self.shape[0]
+        r = zeros(n)
+        f_val = zeros(n, 1)
+        row = 0
+
+        for i in eigen:
+            mul = eigen[i]
+            f_val[row] = f.subs(x, i)
+            if not (f_0.subs(x, i)).is_complex:
+                raise ValueError("Cannot Evaluate the function is not"
+                                 " analytic at some eigen value")
+            val = 1
+            for a in range(n):
+                r[row, a] = val
+                val *= i
+            if mul > 1:
+                mul -= 1
+                coe = [1 for ii in range(n)]
+                deri = 1
+                while(mul >= 1):
+                    row = row + 1
+                    mul -= 1
+                    f_val[row]= derivative[deri].subs(x, i)
```

Comment:
```suggestion
                    f_val[row] = derivative[deri].subs(x, i)
```

Reply:
I'd also expect to raise the error here because there are some cases where the derivative cannot be evaluated for some eigenvalue

e.g
```python3
A = Matrix([[0, -1, -2, 3], [0, -1, -2, 3], [0, 1, 0, -1], [0, 0, -1, 1]])
A.analytic_func(sqrt(x), x)
```

Reply:
Done.